db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1922 @@
+    insertExternalEvent(three);
+    insertExternalEvent(four);
+    insertExternalEvent(five);
+    insertExternalEvent(six);
+    insertExternalEvent(twoC);
+    insertExternalEvent(nine);
+  }
+
+  void insertStandardTypes() throws SQLException {
+    // insert external event type
+    insertExternalEventType(et);
+
+    // insert external source type
+    insertExternalSourceType(st);
+
+    // insert derivation group
+    insertDerivationGroup(dg, st);
+  }
+  //endregion
+
+
+  //region Basic Tests
+  /**
+   * A set of basic tests that verify the general functionality of uploading sources, derived events, the
+   *    derivation_group_comp view, and deletion.
+   * Aside from the derivation_group_comp view test, these tests are not rigorous (more rigorous tests follow in the
+   *    ConstraintsTests class).
+   */
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  class BasicTests {
+    /**
+     * This first test is to verify that a simple upload works, regardless of if its output is correct.
+     * As we do not upload a file we just run the queries that would be run if files were uploaded,
+     * namely tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test.
+     */
+    @Test
+    void uploadWithoutError() {
+      assertDoesNotThrow(() -> upload_source(dg));
+    }
+
+
+    /**
+     * This test uploads data (tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test) and then confirms derived
+     * events associated with the upload perform as expected. All desired keys should be present.
+     * This is an extension of uploadWithoutError.
+     */
+    @Test
+    void basicDerivedEvents() throws SQLException {
+      // upload all source data
+      assertDoesNotThrow(() -> upload_source(dg));
+      try(final var statement = connection.createStatement()) {
+
+        // check that derived events in our prewritten case has the correct keys
+        assertDoesNotThrow(() -> {
+          final var res = statement.executeQuery(
+              // language=sql
+              """
+              SELECT * FROM merlin.derived_events ORDER BY start_time;
+              """
+          );
+
+          final String[] expected_keys = {"1", "9", "3", "5", "6", "2", "8"};
+          compareLists(expected_keys, res, "event_key");
+        });
+      }
+    }
+
+    /**
+     * This test uploads data (tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test) and then confirms
+     * derivation_group_comp view associated with upload performs as expected.
+     * This test alone is rigorous enough for derivation_group_comp view, and is an extension of uploadWithoutError.
+     */
+    @Test
+    void derivationGroupComp() throws SQLException {
+      // upload all source data
+      assertDoesNotThrow(() -> upload_source(dg));
+      try(final var statement = connection.createStatement()) {
+
+        // check that derivation_group_comp has 1 entry, with 4 sources and 7 events
+        assertDoesNotThrow(() -> {
```

Comment:
You don't need to assert that this statement won't throw, as the throwing will already fail the test in this case.